### PR TITLE
Support for XS3 architecture

### PR DESCRIPTION
**Pull request by lucianomartin**

### CODE REVIEW

**Review by ed-xmos — CHANGES_REQUESTED** [review]
Looks OK except for missing copyright date updates in quite a few source file where the license line has been added.

**Review by danielpieczko — APPROVED** [review]
Looks good to me.

**Comment by lucianomartin** [comment]
> Looks OK except for missing copyright date updates in quite a few source file where the license line has been added.

It should be fixed by feature/xs3@4db598c

**Review by ed-xmos — APPROVED** [review]
Looks good

File: lib_i2s/api/i2s.h [hunk 1]
```diff
@@ -209,7 +210,7 @@ void i2s_master(client i2s_callback_if i2s_i,
                 clock bclk,
                 const clock mclk);
 
-#if defined(__XS2A__) || defined(__DOXYGEN__)
+#if defined(__XS2A__) || defined(__XS3A__) || defined(__DOXYGEN__)
 
 /** I2S frame-based master component **for xCORE200 only**
```
Comment (xross) [inline lib_i2s/api/i2s.h:215]: Comment out of date

File: lib_i2s/src/i2s_frame_master_impl.h [hunk 2]
```diff
@@ -1,5 +1,6 @@
-// Copyright (c) 2016-2020, XMOS Ltd, All rights reserved
-#if defined(__XS2A__)
+// Copyright (c) 2016-2021, XMOS Ltd, All rights reserved
+// This software is available under the terms provided in LICENSE.txt.
+#if defined(__XS2A__) || defined(__XS3A__)
```
Comment (xross) [inline lib_i2s/src/i2s_frame_master_impl.h:3]: I'd future proof this with not defined(__XS1A__)

Comment (ed-xmos) [reply lib_i2s/src/i2s_frame_master_impl.h:3]: This is better although we have, rightly or wrongly, used "defined(__XS2A__) || defined(__XS3A__)" everywhere else so will keep this for now as we are super short on time.

File: lib_i2s/api/i2s.h (hunk 1 above)
Comment (ed-xmos) [reply lib_i2s/api/i2s.h:215]: Don't understand?

Comment (xross) [reply lib_i2s/api/i2s.h:215]: **for xCORE200 only** should be xcore.ai and xcore200 I assume

File: lib_i2s/src/i2s_frame_master_impl.h (hunk 2 above)
Comment (xross) [reply lib_i2s/src/i2s_frame_master_impl.h:3]: Sure, just thought I'd save you some effort making all the same mods when XS4 comes out ;)

**Pull request merged**